docs/writing-analysis-jobs.md

```diff
@@ -0,0 +1,126 @@
+
+
+## Example job: which of my EC2 instances is accessible to any host on the internet?
+The easiest way to learn how to write an Analysis Job is through an example.  One of the Analysis Jobs that we've included by default in Cartography's source tree is [`cartography/data/jobs/analysis/aws_ec2_asset_exposure.json`](https://github.com/lyft/cartography/blob/master/cartography/data/jobs/analysis/aws_ec2_asset_exposure.json).  This tutorial covers only the EC2 instance part of that job, but after reading this you should be able to understand the other steps in that file.
+
+### Our goal
+After ingesting all our AWS data, we want to explicitly mark EC2 instances that are accessible to the public internet - a useful thing to know for anyone running an internet service.  If any internet-open nodes are found, the job will add an attribute `exposed_internet = True` to the node.  This way we can easily query to find the assets later on and take remediation action if needed.
+
+But how do we make this determination, and how should we structure the job?
+
+
+### The logic in plain English
+We can use the following facts to tell if an EC2 instance is open to the internet:
+
+1. The EC2 instance is a member of a Security Group that has an IP Rule applied to it that allows inbound traffic from the 0.0.0.0/0 subnet.
+2. The EC2 instance has a network interface that is connected to a Security Group that has an IP Rule applied to it that allows inbound traffic from the 0.0.0.0/0 subnet.
+
+The graph created by Cartography's sync process already has this information for us; we just need to run a few queries to properly to mark it with `exposed_internet = True`!  This example is complex but we hope that this exposes enough Neo4j concepts to help you write your own queries.
+
+
+### Translating the plain-English logic into Neo4j's Cypher syntax
+We can take the ideas above and use Cypher's declarative syntax to "sketch" out this graph path.
+
+1. _The EC2 instance is a member of a Security Group that has an IP Rule applied to it that allows inbound traffic from the 0.0.0.0/0 subnet._
+
+    In Cypher, this is 
+
+    ```
+    MATCH 
+    (:IpRange{id: '0.0.0.0/0'})-[:MEMBER_OF_IP_RULE]->(:IpPermissionInbound)
+    -[:MEMBER_OF_EC2_SECURITY_GROUP]->(group:EC2SecurityGroup)
+    <-[:MEMBER_OF_EC2_SECURITY_GROUP]-(instance:EC2Instance)
+    
+    SET instance.exposed_internet = true, 
+        instance.exposed_internet_type = coalesce(instance.exposed_internet_type , []) + 'direct';
+    ```
+    In the `SET` clause we add `exposed_internet = True` to the instance.  We also add a field for `exposed_internet_type` to denote what type of internet exposure has occurred here.  You can read the [documentation for `coalesce`](https://neo4j.com/docs/cypher-manual/current/functions/scalar/#functions-coalesce), but in English this last part says "mark `exposed_internet_type` as `direct` unless this field was already previously defined on this instance.
```

Comment:
> "mark `exposed_internet_type` as `direct` unless this field was already previously defined on this instance.

It's more like "add 'direct' to the list of ways this instance is exposed to the Internet".